### PR TITLE
Add a devcontainer

### PR DESCRIPTION
**Pull request by RyanBrushett**
### Motivation

This is neat! It will be helpful for new contributors to get an environment quickly, they can edit in the browser, and I made this change from a devcontainer on VSCode so that's cool. Seems nice enough, albeit slow on an M1 mac. I'll do a little review to explain some pieces.

### Implementation

See review comments. TL;DR:
- Pull an amd64 container. This is needed because Sorbet can't install on linux/aarch64. Forcing amd64 means those folks can still bundle install and work.
- I removed any/all node configuration.

### Tests

I used it. I encourage you to try it out too! [Check out their docs to get started](https://code.visualstudio.com/docs/remote/containers-tutorial).

### CODE REVIEW

File: .devcontainer/base.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# [Choice] Ruby version (use -bullseye variants on local arm64/Apple Silicon): 3, 3.0, 2, 2.7, 2.6, 3-bullseye, 3.0-bullseye, 2-bullseye, 2.7-bullseye, 2.6-bullseye, 3-buster, 3.0-buster, 2-buster, 2.7-buster, 2.6-buster
+ARG VARIANT=3-buster
```
[the remark this answers is not in the document]
Comment (paracycle) [reply .devcontainer/base.Dockerfile:2]: Do we have the option of an Ubuntu distro or is it only Debian?

[the remark this answers is not in the document]
Comment (RyanBrushett) [reply .devcontainer/base.Dockerfile:2]: We could use Ubuntu, yes 👍 Would that be preferred?

File: .devcontainer/base.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    && apt-get purge -y imagemagick imagemagick-6-common \
+    # Install common packages, non-root user, rvm, core build tools
+    && bash /tmp/library-scripts/common-debian.sh "${INSTALL_ZSH}" "${USERNAME}" "${USER_UID}" "${USER_GID}" "${UPGRADE_PACKAGES}" "true" "true" \
+    && bash /tmp/library-scripts/ruby-debian.sh "none" "${USERNAME}" "true" "true" \
+    && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/*
+
+ # Remove library scripts for final image
+RUN rm -rf /tmp/library-scripts
+
+
```
Comment (Morriar) [inline .devcontainer/base.Dockerfile:27]:
```suggestion
```

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+			"VARIANT": "3-buster"
+		}
+	},
+
+	// Set *default* container specific settings.json values on container create.
+	"settings": {},
+
+	// Add the IDs of extensions you want installed when the container is created.
+	"extensions": [
+		"rebornix.Ruby"
```
Comment (Morriar) [inline .devcontainer/devcontainer.json:15]: Should we add https://marketplace.visualstudio.com/items?itemName=sorbet.sorbet-vscode-extension?

**Comment by RyanBrushett** [comment]
I forgot about it and now this is broken locally. I'll have to dig back into it.

**Pull request closed without merging**